test/test-helpers.js

```diff
@@ -372,31 +373,29 @@ export const standardXHRResponse = function(request, data) {
     manifestName = request.url;
   }
 
+  const isContainerRequest = request.mimeTypeOverride === 'text/plain; charset=x-user-defined';
+
   if (/\.m3u8?/.test(request.url)) {
     contentType = 'application/vnd.apple.mpegurl';
   } else if (/\.ts/.test(request.url)) {
     contentType = 'video/MP2T';
   } else if (/\.mpd/.test(request.url)) {
     contentType = 'application/dash+xml';
-  } else if (request.responseType === 'arraybuffer') {
+  } else if (request.responseType === 'arraybuffer' || isContainerRequest) {
     contentType = 'binary/octet-stream';
   }
 
   if (!data) {
     data = testDataManifests[manifestName];
   }
 
-  const isTypedBuffer = data instanceof Uint8Array || data instanceof Uint32Array;
-  let response;
+  // default to a uint8array for some old tests.
+  // This may be a good target to clean up in the future.
+  let response = data || new Uint8Array(1024);
 
-  if (isTypedBuffer) {
-    response = data.buffer;
-  } else if (!data) {
-    // A placeholder is used for some old tests. This may be a good target to clean up in
-    // the future.
-    response = new Uint8Array(1024).buffer;
-  } else {
-    response = data;
+  if (isTypedArray(response)) {
+    // a string for container requests or a buffer for non-container requests
+    response = isContainerRequest ? bytesToString(response) : response.buffer;
```

Comment:
container requests are `byte` strings, so that we can work-around the fact that xhr does not add to `request.response` until the request is done, it only adds to `request.responseText`, which throws an error when accesed if the `responseType` is `arraybuffer` like normal.